apps/docs/content/guides/auth/auth-hooks/before-user-created-hook.mdx

```diff
@@ -0,0 +1,488 @@
+---
+id: 'before-user-created-hook'
+title: 'Before User Created Hook'
+subtitle: 'Prevent unwanted signups by inspecting and rejecting user creation requests'
+---
+
+This hook runs before a new user is created. It allows developers to inspect the incoming user object and optionally reject the request. Use this to enforce custom signup policies that Supabase Auth does not handle natively - such as blocking disposable email domains, restricting access by region or IP, or requiring that users belong to a specific email domain.
+
+The hook is triggered any time a new user would be created - including traditional email signups (signUp()), OAuth flows (signIn(\{ provider \})), SSO (signInWithSSO()), IdP-initiated SAML flows, and anonymous sign-ins. Before the user is inserted into the auth.users table, the payload is passed to your configured hook endpoint for evaluation.
```

Comment:
Maybe this would read better if you don't exhaustively list all of the places where it runs?

```suggestion
The hook is triggered any time a new user would be created - including traditional email signups (signUp()), OAuth flows (signIn(\{ provider \})), SSO (signInWithSSO()), IdP-initiated SAML flows, and anonymous sign-ins. Before the user is inserted into the `auth.users` table, the payload is passed to your configured hook endpoint for evaluation.
```

Reply:
I've updated the text.